src/collectd-nagios.c

```diff
@@ -98,6 +98,8 @@ static char *socket_file_g;
 static char *value_string_g;
 static char *hostname_g;
 
+static char range_warning_s[512] = "";
```

Comment:
Wouldn't dynamically assigning the string length be a better option? 
I don't know whether this is available to us... but: https://linux.die.net/man/3/asprintf
See later.

Reply:
I see there's usage of this in [write_sensu](https://github.com/collectd/collectd/blob/main/src/write_sensu.c#L45). I could move that to [utils/common/common.c](https://github.com/collectd/collectd/blob/main/src/utils/common/common.c) what do you think ?

Reply:
Define it globally:
```
char* range_critical_s;
char* range_warning_s;
```